utils/rpms/daos.rpmlintrc

```diff
@@ -42,6 +42,15 @@ addFilter("daos-client-tests\.x86_64: W: files-duplicate \/usr\/lib\/daos\/TESTI
 # per Ashley, we want to supply these even though they cannot be debugged
 addFilter("E: static-library-without-debuginfo \/usr\/lib64\/lib(dfuse|ioil)\.a")
 
+# We want python scripts to use correct python command so they can be invoked properly with
+# virtual environments enabled.
+addFilter("E: env-script-interpreter /usr/lib/daos/TESTING/ftest/cart/cart_logtest.py /usr/bin/env python3")
+addFilter("E: env-script-interpreter /usr/lib/daos/TESTING/ftest/config_file_gen.py /usr/bin/env python3")
+addFilter("E: env-script-interpreter /usr/lib/daos/TESTING/ftest/launch.py /usr/bin/env python3")
+addFilter("E: env-script-interpreter /usr/lib/daos/TESTING/ftest/slurm_setup.py /usr/bin/env python3")
+addFilter("E: env-script-interpreter /usr/lib/daos/TESTING/ftest/verify_perms.py /usr/bin/env python3")
+addFilter("E: env-script-interpreter /usr/bin/daos_storage_estimator.py /usr/bin/env python3")
```

Comment:
If we distribute this one, [this cannot be used per upstream's packaging requirements](https://docs.fedoraproject.org/en-US/packaging-guidelines/Python_201x/#:~:text=Packages%20in%20Fedora%20MUST%20NOT,a%20dependency%20of%20a%20package.)

Reply:
Honestly, I have no idea what the best thing to do here, python packaging in rpms seems totally broken.  We might as well remove this change as the #! line gets re-written at rpm install time anyway, the only way I could get any of this to work was to invoke the code as "python launch.py" which is obviously a change to user behaviour.

I wonder if we'd be better off just removing the #! lines and making the files mode 644.  Seems totally backwards but the rpm restrictions seem designed to make it unusable.

Reply:
> Seems totally backwards but the rpm restrictions seem designed to make it unusable.

On the contrary, RPM packaging is striving to make the result work (consistently) with how python is packaged and distributed on the distribution.

The problem we have is that we are trying to break that convention and have things that are packaged for the distro work with things that are not packaged for the distro.  Herein starts down the path to madness.

Reply:
Like you say the fundamental problem here is that we're packaging python code that depends on other python code that isn't rpm packaged so "there be dragons" is expected however rpm seems to be going out of it's way to prevent us from doing that (with good reason).

Reply:
> Like you say the fundamental problem here is that we're packaging python code that depends on other python code that isn't rpm packaged

What's not packaged that we are depending on?  Maybe the solution is to package whatever that is.

> however rpm seems to be going out of it's way to prevent us from doing that (with good reason).

In what way is RPM preventing us?

> the #! line gets re-written at rpm install time anyway

I'm not aware of this.  Can you expand on your findings in this regard?  Such a thing would cause `rpm -V` to report a massive amount of failure which I have never seen.

> I wonder if we'd be better off just removing the #! lines and making the files mode 644. Seems totally backwards but the rpm restrictions seem designed to make it unusable

In what way?  I think lots of python package ship with _executable_ (i.e. 755 with a `#! /usr/bin/python3` shebang.

I do question whether `daos-devel` is the right place for pydaos.  Maybe it needs it's own sub-package properly packaged as a python module.  It's difficult to say when I don't really understand it's use.